### ISSUE TITLE
Issues with mounts in v2.1.0

### ISSUE THREAD

**Issue by deedeefink** [TechPerplexed/Gooby#36]
I made a fresh install of Ubuntu 18.04.1 and Gooby 2.1.0 and I'm experiencing stability issues with the Rclone/MergerFS  mount. Basically it drops when I do a System Clean and I would have to reboot to get it online again. However, reboot does not always help and it seems overall that the mount is very unstable and not persistent.

Not sure how I can provide more info for debugging, but please let me know .

Rclone setup is:
[Gdrive]
type = drive
scope = drive
root_folder_id = 1***hidden***n
token ={"access_token":"***hidden***","token_type":"Bearer","refresh_token":"1/***hidden***","expiry":"2018-11-12T12:18:37.767056944+01:00"}

**Comment by deedeefink** [comment]
When I navigate via CLI, I can access `/mnt/rclone` and my files show up, but when I navigate to `/mnt/google/` the folder is empty. 

When I navigate via Plex to add a library, the folder `/Media` is empty

**Comment by nesbcn** [comment]
The exact same thing happened to me yesterday. A reboot seemed to fix things, will watch out if it happens again.

**Comment by TechPerplexed** [comment]
Hmmm a  few of us have run it on Ubuntu 18 for several months. It's solid as a rock, I wouldn't release an unstable product (at least not willingly).

How often do you experience a drop of the mount? Is it consistent with certain actions?

 If it's truly unstable for you you can always go back to the old Rclone only solution, but I'm hoping this can be sorted :)

**Comment by TechPerplexed** [comment]
From what you're telling me, it's just MergerFS that is giving you trouble right? As in: /mnt/rclone is still showing your files, but /mnt/google is empty?

When that happens, please type

`sudo systemctl status mnt-google.mount`

What does it report back to you?

(you can escape that with Ctrl-Z)

**Comment by deedeefink** [comment]
Here is what MC is reporting. Again. It only happens after System Clean:

plexuser@Ubuntu-1804-bionic-64-minimal:~$ sudo mc
Directory /tmp/mc-root is not owned by you
Temporary files will be created in /tmp
Press any key to continue...

**Comment by TechPerplexed** [comment]
Hmmm interesting... I researched the message and it dates back to 2012!

You said it only happens after a system clean - but not when you reboot right? Or how do you get the mount to come back at all?

Can you try this please:

```
sudo systemctl stop gooby.service gooby-rclone.service gooby-find.service mnt-google.mount
sudo systemctl disable gooby.service gooby-rclone.service gooby-find.service mnt-google.mount
```

Check if both /mnt/rclone and /mnt/google are empty before you proceed... (they should be) then

```
sudo chown -R $USER:$USER /tmp /mnt
sudo systemctl enable gooby.service gooby-rclone.service gooby-find.service mnt-google.mount
sudo systemctl daemon-reload
sudo systemctl start gooby.service
```

First check if your mounts are there (both /mnt/rclone and /mnt/google, there should also be a /mnt/uploads/Downloads present)

Then run another system clean - did the problem come back?

**Comment by deedeefink** [comment]
Yes, only after system clean

1.  Stopped service = mounts disappear 
2.  Ran commands = mounts are back, also download folder
3.  Ran System Clean= All is working :-)

However, I still get the error with MC as mentioned above after step 2

**Comment by TechPerplexed** [comment]
Ok if everything is working properly I wouldn't worry about that particular error.... the real question is of course: will it keep on working?! Please let me know 😸

**Comment by deedeefink** [comment]
Wait... I were too trigger happy.
After the system clean the mounts are back, but:

- Plex URL not working. I get Chrome privacy error
- Sonarr 502 Bad Gateway
- Radarr, Jackett, Netdata, Tautulli and OMBI are working just fine

A reboot fixes this.

**Comment by deedeefink** [comment]
I'll close now and cross my fingers :-)

**Comment by deedeefink** [comment]
Hmmm, something is not right.
I still have issues which a reboot cannot fix. Now Rclone service is also down, also after reboot.

I'll reset the server and reinstall ubuntu / gooby and let you know if the issues persist afterwards

**Comment by TechPerplexed** [comment]
@deedeefink I just did a system clean and I ran into the same issue as yourself! I made a change in the script, please update Gooby and see if that sorts it for you - once again apologies it's causing you grief!

**Comment by TechPerplexed** [comment]
Mind is suddenly behaving weird too... It's not you. I'll look into this tomorrow!

**Comment by deedeefink** [comment]
Pheww. I was going crazy...
Did a fresh install of ubuntu and Gooby and so far it seems to work as intended. Hopefully that will be the end of this issue :-)

**Comment by deedeefink** [comment]
This also happens if i remove an app/container, so I would assume there is still an issue with permissions.

● mnt-google.mount - MergerFS overlay writable storage
   Loaded: loaded (/etc/systemd/system/mnt-google.mount; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Tue 2018-11-13 14:24:14 CET; 4min 38s ago
    Where: /mnt/google
     What: ${UPLOADS}:${RCLONEMOUNT}
  Process: 5105 ExecUnmount=/bin/umount /mnt/google -c (code=exited, status=0/SUCCESS)
  Process: 5903 ExecMount=/bin/mount ${UPLOADS}:${RCLONEMOUNT} /mnt/google -t fuse.mergerfs -o use_ino,hard_remove,auto_cache,sync_read,allow_other,category.action=all,category.crea
    Tasks: 0 (limit: 4915)
   CGroup: /system.slice/mnt-google.mount

Nov 13 14:24:14 Ubuntu-1804-bionic-64-minimal systemd[1]: Mounting MergerFS overlay writable storage...
Nov 13 14:24:14 Ubuntu-1804-bionic-64-minimal mount[5903]: fuse: missing mountpoint parameter
Nov 13 14:24:14 Ubuntu-1804-bionic-64-minimal systemd[1]: mnt-google.mount: Mount process exited, code=exited status=1
Nov 13 14:24:14 Ubuntu-1804-bionic-64-minimal systemd[1]: mnt-google.mount: Failed with result 'exit-code'.
Nov 13 14:24:14 Ubuntu-1804-bionic-64-minimal systemd[1]: Failed to mount MergerFS overlay writable storage.

**Comment by bdschuster** [comment]
This is really making me think that 18.04.1 has changed some things. I kept having issues like this, I reverted back to 18.04.0 and no issues anymore. It also seems like they changed the repository as well since you can no longer access some common tools like fail2ban from the repository any longer in 18.04.1.

**Comment by deedeefink** [comment]
Could very well be. In this case @TechPerplexed should experience the same issues as I believe we're both using Hetzner who only offers 16.04.5 and 18.04.1 as default Ubuntu images. 

But I'll consider to downgrade

**Comment by TechPerplexed** [comment]
I'm sure we can get around this, no worries... I'm pretty much sure I know where the problem lies, so all that needs to be fine now is implement the correct solution!

**Comment by TechPerplexed** [comment]
Guys, it's definitely something to do with whatever change happened between v18.04.0 and 18.04.1 - in fact I can reproduce the issue 100% now. That also explains why it can flawlessly for months before going live.

The positive news is that I am pretty sure I have a solution... which I'm going to implement in a test branch and will probably ask you to test before going live with it.

If you want the technical details:

Currently Gooby creates a single service that calls upon Rclone and MergerFS and boots them up one by one in sequence through the script. This worked like a charm in the past, but it's broken on the latest patch. It's not even intermittent any longer, it just crashes.

The solution is to go back to two separate scripts, which is the conventional way of running scripts anyway and seems to work fine.

I hope to have more to have more news tomorrow and a place for you to test my theories :)

**Comment by TechPerplexed** [comment]
Lol, I have no idea, but frankly it's driving me nuts... I can make it work now on one machine (Hetzner) but not on my regular dedi... but eh I'm not going to give up that easily! I love using MergerFS because it prevents so many bans and other problems. 

New day tomorrow with some fresh and innovative thoughts - hopefully 😆

**Comment by TechPerplexed** [comment]
Well I just had a bit of a brainwave.... can you try updating Gooby to the latest version and see if that works better? If not, I WILL use tomorrow for fresh, creative and brilliant thoughts LOL

**Comment by bdschuster** [comment]
Just update? Or do a system cleanup after as well?

**Comment by TechPerplexed** [comment]
Update, then see if a system cleanup or reboot kills the mount (if it does, I'm gonna kill Ubuntu!)

**Comment by TechPerplexed** [comment]
BAHHH HUMBUG!!!!!!!!!!!!!!!!!!!!!! STILL no good!!! Weirdly enough a manual 

`sudo systemctl start gooby`

Will bring the mount back instantly (at least for me), but it's supposed to come back in the SCRIPT! GRR, ah well, tomorrow!

**Comment by TechPerplexed** [comment]
@deedeefink Do you still have issues with the Rclone mount, or just MergerFS? 

@bdschuster Yes, that is exactly what is happening to me too now. Yet it worked fine for me until yesterday - then suddenly all hell broke loose. The mount is stable once it's up, but a reboot or system clean WILL break it 100% now. Crazy!

**Comment by bdschuster** [comment]
I know this sounds crazy...but try adding a pause...maybe 10 seconds to start with, right above the the start gooby command in the script.

**Comment by deedeefink** [comment]
> @deedeefink Do you still have issues with the Rclone mount /mnt/rclone, or just MergerFS /mnt/google?

So, 
1. I updated Gooby and verified that I got your last commit 
2. Performed a system Clean = Both Rclone and MergerFS went down
3. I rebooted twice = MergerFS is mounted, but no content as Rclone is down
4. Ran `sudo systemctl start gooby` = MergerFS is mounted, but no content as Rclone is down

Status are:
**gooby-rclone.service** - Mount Google Drive (rclone)
   Loaded: loaded (/etc/systemd/system/gooby-rclone.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2018-11-14 13:14:06 CET; 1min 3s ago
  Process: 17519 ExecStartPre=/bin/mkdir -p ${RCLONEMOUNT} (code=exited, status=1/FAILURE)

**mnt-google.mount** - MergerFS overlay writable storage
   Loaded: loaded (/etc/systemd/system/mnt-google.mount; enabled; vendor preset: enabled)
   Active: active (mounted) since Wed 2018-11-14 13:11:11 CET; 5min ago
    Where: /mnt/google
     What: /mnt/uploads
  Process: 6398 ExecUnmount=/bin/umount /mnt/google -c (code=exited, status=0/SUCCESS)
  Process: 6479 ExecMount=/bin/mount ${UPLOADS}:${RCLONEMOUNT} /mnt/google -t fuse.mergerfs -o use_ino,hard_remove,auto_cache,sync_read,allow_other,category.action=all,category.crea
    Tasks: 4 (limit: 4915)
   CGroup: /system.slice/mnt-google.mount
           └─6493 mergerfs /mnt/uploads:/mnt/rclone /mnt/google -o rw,use_ino,hard_remove,auto_cache,sync_read,allow_other,category.action=all,category.create=ff,dev,suid

**Comment by TechPerplexed** [comment]
Thanks guys, I'll try to tackle this with renewed zeal today. WE SHALL LICK THIS! Or my name is not Tech Perplexed - oh wait it's not 👩

**Comment by TechPerplexed** [comment]
@deedeefink one more question for you: when you do a fresh install of Rclone, do both mounts ALWAYS come up properly? In other words, do your woes only happen after a reboot/system clean?

**Comment by deedeefink** [comment]
> @deedeefink one more question for you: when you do a fresh install of Rclone, do both mounts ALWAYS come up properly? In other words, do your woes only happen after a reboot/system clean?

Yes that is correct. But, I only removed Rclone by reinstalling Ubuntu and then install Rclone on a fresh box. I did not try `sudo apt-get remove rclone` and the reinstall Rclone. Do you want me to try this?

**Comment by TechPerplexed** [comment]
Would you please try to uninstall it from the menu and then reinstall it (again from the menu)? No need to delete your config... just uninstall and reinstall. I'm trying to figure out if a fresh installation is behaving different from the system clean and then take it from there :)

**Comment by deedeefink** [comment]
Yes that works. mounts are back up again :-)

**Comment by TechPerplexed** [comment]
SUPER, ok, and I made a few changes to the uninstall script, would you please update Gooby and see if it's still behaving well? (And thank you for testing this with me)

**Comment by deedeefink** [comment]
Yes, still works
(and no worries. Happy to help)

**Comment by TechPerplexed** [comment]
Ok, the update is live.... go for it 🙈

**Comment by deedeefink** [comment]
Is it working on your serve?
Can there be something else that is interfering on my side?

**Comment by TechPerplexed** [comment]
Yeah it just behaved weird on mine too, it's definitely not you. It behaved well for 2 reboots, then suddenly my mount dropped too. This is unbelievable... I mean if uninstalling/reinstalling works, then there isn't a reason in the world why it wouldn't behave the same in an almost identical script.

Ok, back to the drawing board..... ackpff.

**Comment by deedeefink** [comment]
As mentioned a couple of time I like to use Midnight Commander and it's my first go-to tool to see if my mount is available. After each System Clean I get an error starting MC:

```
Directory /tmp/mc-root is not owned by you
Temporary files will be created in /tmp
Press any key to continue...
``` 

In the system clean script you're resetting permissions. As I remember this is a relatively new addition to the script. Could it be that this resetting permissions are impacting the mounts?

I'm thinking this as I typically use `sudo mc` to launch Midnight Commander. When I use just `mc` I do not get the permission error. I mean, it's weird that I get the permissions error when I used `sudo`

**Comment by TechPerplexed** [comment]
Thank you, I must look into that deeper then. Now here is the REALLY weird issue:

The two scripts are basically identical now. In fact I went one further and added the command to uninstall and reinstall Rclone (on the test script, not live) - just so it matches the two scripts 100%.

Yet it's STILL not coming back properly for me.

The resetting permissions were added because I wanted to match the working environment to a T (but if you see something I don't, all suggestions welcome!!!)

All this started when you started this thread - it's all your fault (TEASING) 😋 

Seriously though I do feel we're very close... if one script works, with another one it MUST be possible to make to toe the line too right?

**Comment by deedeefink** [comment]
I accept fully responsibility ;-)

**Comment by bdschuster** [comment]
If he didn't open, I was going too 😃😃 ... And sorry I haven't been around, been dealing with medical issues.

**Comment by deedeefink** [comment]
@bdschuster welcome back. Hope things are okay?

**Comment by bdschuster** [comment]
I'll live (hopefully) lol

**Comment by deedeefink** [comment]
I'll soon be signing out for tonight and leave you too it. Will test in the morning.

**Comment by TechPerplexed** [comment]
@deedeefink sleep well, and @bdschuster take care, hope you're all better soon <3

**Comment by TechPerplexed** [comment]
Ok, here is a test environment for you guys with a new service (Rclone and MergerFS each separately). To install:

```
sudo rm -r /opt/Gooby
sudo git clone -b rcleantest https://github.com/TechPerplexed/Gooby /opt/Gooby
sudo chmod +x -R /opt/Gooby/install
sudo chmod +x -R /opt/Gooby/menus
sudo chmod +x -R /opt/Gooby/scripts/cron
sudo rsync -a /opt/Gooby/install/gooby /bin
sudo chmod 755 /bin/gooby
```

Then uninstall Rclone from the menu and reinstall (without deleting the config).

Let's see how that goes! Don't update Gooby from the menu as that will take you back to the stable (cough, make that no-longer-stable) branch.

I rebooted and updated quite a few times and it seems alright to me... very curious how it goes for you!

**Comment by HitsvilleUK** [comment]
3 cleanups and 4 reboots performed and both /mnt/google & /mnt/clone come back each time.

**Comment by deedeefink** [comment]
After 2x system clean and 2x reboot both Rclone and MergerFS mounts are coming back online... Wuhuuuu.... but (sorry).....theres more......

After my first system clean non of the containers worked and I could not access any apps - But mounts were up though :-). I rebooted and checked mounts. All okay. But still Docker were not working. I performed a new system clean but it never completed as it got stuck at "Bringing services back online". It gave me time to notice the following error:

```
Taking containers down
Removing network docker_default
WARNING: Network docker_default not found.
```

I wanted to double check that I could reproduce the above. When I tired, I encountered no problems with docker_network, but the mounts were empty...arrrggghhhh....I tried to system clean and switching back and forth between the "stable" and test version for the N number of times and:

- Sometime it works
- Sometimes I get stuck at "Bringing services back online"
- Sometime docker_default is not working
- Sometimes mounts are failing
- Sometimes I smile and sometimes I cry

@nesbcn, @bdschuster and @HitsvilleUK can you please check. I'm loosing track of all the things I tried and need a second opinion

**Comment by HitsvilleUK** [comment]
Interesting @deedeefink 
After all my system cleans all the containers loaded fine. I just double checked again now and they're all up including my custom containers, 
I suddenly started getting issues a day or so ago scanning my TV shows into Emby and even that's working now, Sorry I cant be of more help.

**Comment by deedeefink** [comment]
I tried a system clean again. It completed and the mounts are working, but the docker_network is not responding and I cannot access my apps. Also after a reboot.

I then make another system clean and get this error again:
```
Taking containers down
Removing network docker_default
WARNING: Network docker_default not found.
```
And "Bringing services back online" gets stuck

@TechPerplexed / @all I'll fly off to the beach now , so I might not be online for testing that much the coming days.

**Comment by TechPerplexed** [comment]
@deedeefink first of all, have a wonderful vacation (or holiday, depending on where you live 🤣 )

I'm glad the mount problem seems to be resolved... tentatively of course.

As for the message "docker_default not found", this is normal and should not have any other effect (it's just to make sure everything is cleaned out properly). It is also normal for the containers not to always show the "Ok" message (that always makes me nervous too).

HOWEVER your system should come back of course... unfortunately I can't reproduce it myself. For the record there have been absolutely no changes in the network settings or the rest of the script - I have only been "messing" with the mount settings.

I suggest that if the mount is fine (in the new test setting), but if you continue to have problems with the docker_network not responding, that you create a new issue so we can start troubleshooting that separately...

If you guys all confirm that the new services setting solves the mount problem I'll push that to stable - and then we can start afresh with new problems (YIKES I hope not...)

Today I am going to take a day off too, because three days of non stop troubleshooting sort of made me neglect all my regular activities, heh. Tomorrow I'll be back with renewed vim and vigor!

**Comment by HitsvilleUK** [comment]
Yeah I should have said....in case it's relevant to deedeefink.
The containers take a couple of minutes to load (which is fine) Plex and and Emby especially a minute or so longer still.
When I tested I think I got the 502 (or whichever) error. Leaving it a few minutes all was good.

**Comment by TechPerplexed** [comment]
Ok my friendlies, this is what I'm going to do... since it seems that is new script has solved the mount problems, I'll push it to the "master" repository tomorrow. If you wish you can update Gooby from the menu (option C - A) and go with that (tomorrow that is, not just yet).

However, we're not done yet with the test script.... with the help of my good friend kelinger we have made more improvements to the mount check, basically that it won't start any of the containers until it verifies that both the Rclone AND the MergerFS mount are in place and running.

After I push this test repository to the live environment I'll update the test place with the improvements he has made - which will then be pushed soon afterwards hopefully after your approval :)

But all that is for tomorrow, today was strictly a script free day for me 😴

**Comment by TechPerplexed** [comment]
Done! The version you guys tested is now live - and I have updated the test environment with the built in mount checks (to make sure they are really up before proceeding). Hopefully this can go live by the weekend 👍

**Comment by bdschuster** [comment]
lol...i will install and check everything out shorty and report back...get some rest.

**Comment by nesbcn** [comment]
My system is also stuck at this:

```
mountpoint: /mnt/rclone: No such file or directory
Waiting on mounts to be created...
mountpoint: /mnt/rclone: No such file or directory
Waiting on mounts to be created...
```
this is after updating gooby and system clean. Any suggestion?

**Comment by TechPerplexed** [comment]
How long does it stay stuck in that loop? Forever or a certain amount of time?

Also lots of suggestions in the post above you :)

**Comment by nesbcn** [comment]
I tried everything in the post, still looping forever:

```
The latest version of rclone rclone v1.44 is already installed.


Cleaning mount leftovers

Job for rclonefs.service failed because the control process exited with error code.
See "systemctl status rclonefs.service" and "journalctl -xe" for details.
Job for mergerfs.service failed because the control process exited with error code.
See "systemctl status mergerfs.service" and "journalctl -xe" for details.
mountpoint: /mnt/rclone: No such file or directory
mountpoint: /mnt/google: No such file or directory
Waiting on mounts to be created...
mountpoint: /mnt/rclone: No such file or directory
mountpoint: /mnt/google: No such file or directory
Waiting on mounts to be created...
mountpoint: /mnt/rclone: No such file or directory
mountpoint: /mnt/google: No such file or directory
Waiting on mounts to be created...
mountpoint: /mnt/rclone: No such file or directory
mountpoint: /mnt/google: No such file or directory
Waiting on mounts to be created...
mountpoint: /mnt/rclone: No such file or directory
mountpoint: /mnt/google: No such file or directory
Waiting on mounts to be created...
mountpoint: /mnt/rclone: No such file or directory
mountpoint: /mnt/google: No such file or directory
Waiting on mounts to be created...
mountpoint: /mnt/rclone: No such file or directory
mountpoint: /mnt/google: No such file or directory
Waiting on mounts to be created...
mountpoint: /mnt/rclone: No such file or directory
mountpoint: /mnt/google: No such file or directory
Waiting on mounts to be created...
mountpoint: /mnt/rclone: No such file or directory
mountpoint: /mnt/google: No such file or directory
Waiting on mounts to be created...

```

**Comment by TechPerplexed** [comment]
Can you try once more to set the permissions? Problem is I can't reproduce this any longer... testing on two systems and it just seems to work fine for me.

So try this:

sudo systemctl stop mergerfs
sudo systemctl stop rclonefs
sudo systemctl daemon-reload
sudo chown -R $USER:$USER /mnt

Then make sure Gooby is updated and try the cleanup again from the menu...

**Comment by nesbcn** [comment]
Well I didn't try the permissions, I updated Gooby again, rebooted then reinstalled rclone (Beta version this time), then proceeded to system clean. It worked!

**Comment by nesbcn** [comment]
Also, would it be possible to launch rclone mount with `--rc` option? I like to review the stats from time to time, to check everything is going smoothly...

**Comment by deedeefink** [comment]
I tested what you suggested in the above comment https://github.com/TechPerplexed/Gooby/issues/36#issuecomment-439633265

- user_allow_other is uncommented
- I chown /mnt folder

Here is my env.:
```
RCLONESERVICE=Gdrive
RCLONEFOLDER=
RCLONEMOUNT=/mnt/rclone
MOUNTTO=/mnt/google
UPLOADS=/mnt/uploads
GOOGLE=/mnt/google
MEDIA=/mnt/google
MOVIES=/mnt/google/Movies
TV=/mnt/google/TV
DOWNLOADS=/home/plexuser/Downloads
```
Note that I define the root folder in my rclone settings. Don't think it's relevant, but my TV folder is actual TVshows

When i reboot with / without disabled crontab mounts are not working. When I test it, the result is:
```
$ sudo systemctl status gooby-rclone.service  
Unit gooby-rclone.service could not be found.
```
```
$ sudo systemctl status mnt-google.mount
Unit mnt-google.mount could not be found.
```

**Comment by nesbcn** [comment]
Well that was too quick... Mounts seem to be working, now Plex somehow doesn't find the files when transcoding ... I'm looking into it atm

**Comment by deedeefink** [comment]
> sudo systemctl stop mergerfs 
> sudo systemctl stop rclonefs
> sudo systemctl daemon-reload

Followed this, updated Gooby, reinstalled rclone and ran system clean and it worked :-)

I did not add my media just yet, so can't verify @nesbcn Plex problem

**Comment by TechPerplexed** [comment]
> Also, would it be possible to launch rclone mount with `--rc` option? I like to review the stats from time to time, to check everything is going smoothly...

Absolutely! You can edit your rclonefs.service to your liking - it won't be overwritten unless you uninstall/reinstall :)
`sudo nano /etc/systemd/system/rclone.service`

> Well that was too quick... Mounts seem to be working, now Plex somehow doesn't find the files when transcoding ... I'm looking into it atm

Did you restart the containers though? After all we stopped the services without stopping the containers first... 

> Followed this, updated Gooby, reinstalled rclone and ran system clean and it worked :-)

Oh I hope so.... fingers crossed... toes crossed... eyes crossed........

**Comment by HitsvilleUK** [comment]
Yep. I just had real life cause to use a system clean up.
It definitely works for me here too. Nice work Sonia and Kelinger.

**Comment by nesbcn** [comment]
after several system cleanups, everything is still working. 
case closed!
